### PR TITLE
Add `deployOperator` option to the AWS kind E2E environment 

### PR DESCRIPTION
**Pull request by fanny-jiang**
### What does this PR do?

* Add `deployOperator` option to the AWS kind provisioner
* Add 2 examples of e2e tests for deploying the agent using the operator for local kind and AWS kind 

### Motivation

* Enable E2E testing for operator-managed agents

### CODE REVIEW

File: .gitlab/common/test_infra_version.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ variables:
   # and check the job creating the image to make sure you have the right SHA prefix
   TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX: ""
   # Make sure to update test-infra-definitions version in go.mod as well
-  TEST_INFRA_DEFINITIONS_BUILDIMAGES: 4b4112f5f64d
+  TEST_INFRA_DEFINITIONS_BUILDIMAGES: 10486aec63a8
```
Comment (pducolin) [inline .gitlab/common/test_infra_version.yml:7]: **💬 suggestion**
Recommend bumping `test-infra-definitions` through [the automated bump PR](https://github.com/DataDog/datadog-agent/pull/32321), the advantage is it contains the diff, which ease investigations during incidents. You can also cherry-pick the bump commit  and copy the PR description. We'll probably merge the bump PR before US biz hours today, you can rebase on main after that

File: test/new-e2e/pkg/components/operator.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// Copyright 2016-present Datadog, Inc.
+
+package components
+
+import (
+	"github.com/DataDog/test-infra-definitions/components/datadog/operator"
+)
+
+// Operator is a Datadog Operator running in a Kubernetes cluster
+type Operator struct { // nolint:revive, We want to keep the name as <Component>
```
Comment (pducolin) [inline test/new-e2e/pkg/components/operator.go:13]: **💬 suggestion**
Is lint complaining here? It should not, as the package's name is not `operator`
```suggestion
type Operator struct {
```

File: test/new-e2e/pkg/components/operator.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
```
Comment (pducolin) [inline test/new-e2e/pkg/components/operator.go:1]: **💬 suggestion**
Looks like this component is never used, and has no client associated. You can skip it if it's not meant to receive an export from a pulumi output

File: test/new-e2e/pkg/components/operator.go (hunk 2 above)
Comment (fanny-jiang) [reply test/new-e2e/pkg/components/operator.go:12]: Ah thanks, that definitely shouldn't be there

**Review by pducolin — APPROVED** [review]

**Pull request merged**